src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py

```diff
@@ -373,3 +375,59 @@ def test_create_storage_account_with_files_aadds_true(self, resource_group):
 
         self.assertIn('azureFilesIdentityBasedAuthentication', result)
         self.assertEqual(result['azureFilesIdentityBasedAuthentication']['directoryServiceOptions'], 'AADDS')
+
+
+class RoleScenarioTest(LiveScenarioTest):
+    def run_under_service_principal(self):
+        account_info = self.cmd('account show').get_output_in_json()
+        return account_info['user']['type'] == 'servicePrincipal'
+
+
+@api_version_constraint(ResourceType.MGMT_STORAGE, min_api='2019-04-01')
+class RevokeStorageAccountTests(StorageScenarioMixin, RoleScenarioTest, LiveScenarioTest):
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer()
+    def test_storage_account_revoke_delegation_keys(self, resource_group, storage_account):
+        if self.run_under_service_principal():
+            return  # this test delete users which are beyond a SP's capacity, so quit...
+        from datetime import datetime, timedelta
+        import time
+
+        expiry = (datetime.utcnow() + timedelta(hours=1)).strftime('%Y-%m-%dT%H:%MZ')
+        account_info = self.get_account_info(resource_group, storage_account)
+        c = self.create_container(account_info)
+        b = self.create_random_name('blob', 24)
+        local_file = self.create_temp_file(128, full_random=False)
+        self.kwargs.update({
+            'expiry': expiry,
+            'account': storage_account,
+            'container': c,
+            'local_file': local_file,
+            'blob': b,
+            'rg': resource_group
+        })
+        result = self.cmd('storage account show -n {account} -g {rg}').get_output_in_json()
+        self.kwargs['sc_id'] = result['id']
+
+        user = self.create_random_name('testuser', 15)
+        self.kwargs['upn'] = user + '@azuresdkteam.onmicrosoft.com'
+        self.cmd('ad user create --display-name tester123 --password Test123456789 --user-principal-name {upn}')
```

Comment:
so we don't have a command like `wait` for `ad user create`. right?

Reply:
It looks we don't have corresponding command, so us time.sleep(15) to wait.